pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ReaderBuilder.java

```diff
@@ -121,6 +121,13 @@
      */
     ReaderBuilder<T> startMessageId(MessageId startMessageId);
 
+    /**
+     * The initial reader positioning is done by specifying a specific timestamp.
+     * @param timestamp the timestamp where the reader will be initially positioned on
+     * @return the reader builder instance
+     */
+    ReaderBuilder<T> startMessageId(long timestamp);
```

Comment:
I would suggest calling it `startPublishTime` because `timestamp` is not a message id.